homeassistant/components/mealie/coordinator.py

```diff
@@ -22,18 +25,54 @@
 
 WEEK = timedelta(days=7)
 
-type MealieConfigEntry = ConfigEntry[MealieCoordinator]
 
+@dataclass
+class MealieData:
+    """Mealie data type."""
 
-class MealieCoordinator(DataUpdateCoordinator[dict[MealplanEntryType, list[Mealplan]]]):
-    """Class to manage fetching Mealie data."""
+    client: MealieClient
+    mealplan_coordinator: MealieMealplanCoordinator
+    shoppinglist_coordinator: MealieShoppingListCoordinator
+
+
+type MealieConfigEntry = ConfigEntry[MealieData]
+
+
+class MealieDataUpdateCoordinator[_DataT](DataUpdateCoordinator[_DataT]):
+    """Base coordinator."""
 
     config_entry: MealieConfigEntry
 
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        client: MealieClient,
+        update_interval: timedelta,
+    ) -> None:
+        """Initialize the Withings data coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name="",
+            update_interval=update_interval,
+        )
+        self.name = f"Mealie {self.config_entry.unique_id} {name}"
```

Comment:
The unique id is a uuid, so let's not do that in the mealie name as it's not user recognizable